src/is-of-type.ts

```diff
@@ -0,0 +1,99 @@
+import { Action, AnyAction } from './create-action'
+import { ActionCreator } from './create-action-creator'
+import { getType } from './get-type'
+import { castArray } from './utils'
+
+/**
+ * Non-curried function
+ * @example
+ * const increment = createActionCreator('[Counter] increment')
+ * const decrement = createActionCreator('[Counter] decrement')
+ * const reset = createActionCreator(
+ *   '[Counter] reset',
+ *   resolve => (value: number) => resolve(value)
+ * )
+ * isOfType(increment.type, increment()) //=> true
+ * @example
+ * isOfType([increment.type, decrement.type], increment()) //=> true
+ * @example
+ * isOfType(decrement(), increment()) //=> false
+ */
+export function isOfType<
+  TSource extends string | AnyAction | ActionCreator<AnyAction>,
+  TAction extends AnyAction
+>(
+  type: TSource | TSource[],
+  action: TAction
+): action is TAction extends Action<
+  TSource extends string
+    ? TSource
+    : TSource extends AnyAction | ActionCreator<AnyAction>
+    ? TSource['type']
+    : never
+>
+  ? TAction
+  : never
+
+/**
+ * Curried function
+ * @example
+ * const increment = createActionCreator('[Counter] increment')
+ * const decrement = createActionCreator('[Counter] decrement')
+ * isOfType(increment.type)(increment()) //=> true
+ * @example
+ * isOfType([increment.type, decrement.type])(increment()) //=> true
+ * @example
+ * isOfType(decrement())(increment()) //=> false
+ */
+export function isOfType<
+  TSource extends string | AnyAction | ActionCreator<AnyAction>
+>(
+  type: TSource | TSource[]
+): <TAction extends AnyAction>(
+  action: TAction
+) => action is TAction extends Action<
+  TSource extends string
+    ? TSource
+    : TSource extends AnyAction | ActionCreator<AnyAction>
+    ? TSource['type']
+    : never
+>
+  ? TAction
+  : never
```

Comment:
This conditional type used twice in this file, also we can use it in `ofType` RxJS operator. So we can extract it as a helper type in types.ts file named `ExtractAction`.

Reply:
completed! https://github.com/thebrodmann/deox/pull/127/commits/47e6bee3863a4e121b292fb4500bfca49924d6ec